### PR TITLE
:seedling: Unit tests for Gitlab status and WebHooks

### PR DESCRIPTION
**Pull request by naveensrinivasan**
#### What kind of change does this PR introduce?
Included test for Gitlab status and webhook

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: clients/gitlabrepo/webhook_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	responsePath string
+}
+
+func (s stubTripper) RoundTrip(_ *http.Request) (*http.Response, error) {
+	f, err := os.Open(s.responsePath)
+	if err != nil {
+		//nolint:wrapcheck
+		return nil, err
+	}
+	return &http.Response{
```
Comment (spencerschrock) [inline clients/gitlabrepo/webhook_test.go:38]: Need to `defer f.Close()` here as well. I know this was copy-pasted from a similar test I did for the GitHub handler. I forgot it there too :(

File: clients/gitlabrepo/webhook_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+				{
+					ID:             1,
+					Path:           "http://example.com/hook",
+					UsesAuthSecret: true,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:         "invalid webhook",
```
Comment (spencerschrock) [inline clients/gitlabrepo/webhook_test.go:66]: it's not so much the webhook is invalid, but rather that `listWebhooks` returns an error. Should the name match that?

What do you think about: "failure fetching webhooks"

File: clients/gitlabrepo/statuses_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+func Test_listStatuses(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name         string
+		responsePath string
+		want         []clients.Status
+		wantErr      bool
+	}{
+		{
+			name:         "valid webhook",
```
Comment (spencerschrock) [inline clients/gitlabrepo/statuses_test.go:36]: please fix copy/paste mistake

File: clients/gitlabrepo/statuses_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+					State:     "pending",
+					Context:   "bundler:audit",
+					URL:       "https://gitlab.example.com/janedoe/gitlab-foss/builds/91",
+					TargetURL: "https://gitlab.example.com/janedoe/gitlab-foss/builds/91",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:         "invalid webhook",
```
Comment (spencerschrock) [inline clients/gitlabrepo/statuses_test.go:49]: copy/paste typo, but also see comment below on test name.

what do you think about "handles error listing statuses" or something like that?

**Review by spencerschrock — APPROVED** [review]
Thanks

File: clients/gitlabrepo/statuses_test.go (hunk 4 above)
Comment (naveensrinivasan) [reply clients/gitlabrepo/statuses_test.go:49]: 👍

File: clients/gitlabrepo/statuses_test.go (hunk 3 above)
Comment (naveensrinivasan) [reply clients/gitlabrepo/statuses_test.go:36]: 👍

File: clients/gitlabrepo/webhook_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply clients/gitlabrepo/webhook_test.go:66]: 👍

File: clients/gitlabrepo/webhook_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply clients/gitlabrepo/webhook_test.go:38]: 👍

Comment (spencerschrock) [reply clients/gitlabrepo/webhook_test.go:38]: Note: I was wrong. The file close is unnecessary as the response body gets closed by the caller down the line.

**Pull request merged**